### PR TITLE
Added Markdown support

### LINKED ISSUE #16: Markdown support
Add markdown parsing to the note editor. It should support all major features of markdown.

Option A (preferred):
Live parsing. Similar to Slack and Discord, as the user types, parse their text.

Option B:
Add a toggle for the user to change between the editor and a preview.

### PR DESCRIPTION
**Pull request by mstarski**
Added ability to toggle markdown preview of currently edited note. (#16)

### CODE REVIEW

File: src/components/PreviewControls.vue [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  * + * {
+    margin-left: 1rem;
+  }
+}
+
+.control-btn {
+  padding: 0.5rem 2rem;
+  border: 1px solid #fff;
+  color: #fff;
+  border-radius: 10px;
```
Comment (1VinceP) [inline src/components/PreviewControls.vue]: Change border-radius to 3px to maintain consistency

**Review by 1VinceP — CHANGES_REQUESTED** [review]
Looks good functionally. Just a minor change with the button radius (comment attached to code).

However, I noticed there are a lot of code style changes (indent 3 -> 2 spaces, single quotes -> double quotes, etc). I would appreciate if those changes would be reverted.

Also, if you could keep the documentation comments, that would be great. I originally built this to help some others learn Vue and I would like their comments to remain (see #15).

File: src/components/PreviewControls.vue (hunk 1 above)
Comment (mstarski) [reply src/components/PreviewControls.vue]: Done.

**Comment by mstarski** [comment]
Hey! Thank you for reviewing my change. I restored back comments and indentation (although it's weird - i am pretty sure i didnt have these comments from the beginning) as well as changed the buttons' radius. As for code styling - it is quite problematic to contribute to a project without any code configuration files like `.editorconfig` or `.prettierrc` hence the formatting changes. I thought I'd add it here to avoid further problems. I set parameters according to your review. Additionaly I've added a npm script to allow pontential contributors to run their code against prettier and auto fix is necessary. Please see for yourself if it's ok.

**Review by 1VinceP — APPROVED** [review]

**Comment by 1VinceP** [comment]
Yeah, thanks, prettier is a good idea. Everything looks great!

**Pull request merged**